src/gt4py/next/program_processors/runners/dace_iterator/__init__.py

```diff
@@ -155,214 +147,3 @@ def build_sdfg_from_itir(
-    connectivity_tables_data_descriptors: ClassVar[
-        dict[str, dace.data.Array]
-    ] = {}  # symbolically defined
-
-    def __sdfg__(self, *args, **kwargs) -> dace.sdfg.sdfg.SDFG:
-        if "dace" not in self.backend.name.lower():  # type: ignore[union-attr]
-            raise ValueError("The SDFG can be generated only for the DaCe backend.")
-
-        params = {str(p.id): p.type for p in self.itir.params}
-        fields = {str(p.id): p.type for p in self.itir.params if hasattr(p.type, "dims")}
-        arg_types = [*params.values()]
-
-        dace_parsed_args = [*args, *kwargs.values()]
-        gt4py_program_args = [*params.values()]
-        _crosscheck_dace_parsing(dace_parsed_args, gt4py_program_args)
-
-        if self.connectivities is None:
-            raise ValueError(
-                "[DaCe Orchestration] Connectivities -at compile time- are required to generate the SDFG. Use `with_connectivities` method."
-            )
-        offset_provider = (
-            self.connectivities | self._implicit_offset_provider
-        )  # tables are None at this point
-
-        sdfg = self.backend.executor.step.translation.generate_sdfg(  # type: ignore[union-attr]
-            self.itir,
-            arg_types,
-            offset_provider=offset_provider,
-            column_axis=kwargs.get("column_axis", None),
-        )
-        self.sdfg_closure_vars["sdfg.arrays"] = sdfg.arrays  # use it in __sdfg_closure__
-
-        # Halo exchange related metadata, i.e. gt4py_program_input_fields, gt4py_program_output_fields, offset_providers_per_input_field
-        # Add them as dynamic properties to the SDFG
-
-        assert all(
-            isinstance(in_field, SymRef)
-            for closure in self.itir.closures
-            for in_field in closure.inputs
-        )  # backend only supports SymRef inputs, not `index` calls
-        input_fields = [
-            str(in_field.id)  # type: ignore[union-attr]  # ensured by assert
-            for closure in self.itir.closures
-            for in_field in closure.inputs
-            if str(in_field.id) in fields  # type: ignore[union-attr]  # ensured by assert
-        ]
-        sdfg.gt4py_program_input_fields = {
-            in_field: dim
-            for in_field in input_fields
-            for dim in fields[in_field].dims  # type: ignore[union-attr]
-            if dim.kind == common.DimensionKind.HORIZONTAL
-        }
-
-        output_fields = []
-        for closure in self.itir.closures:
-            output = closure.output
-            if isinstance(output, itir.SymRef):
-                if str(output.id) in fields:
-                    output_fields.append(str(output.id))
-            else:
-                for arg in output.args:
-                    if str(arg.id) in fields:  # type: ignore[attr-defined]
-                        output_fields.append(str(arg.id))  # type: ignore[attr-defined]
-        sdfg.gt4py_program_output_fields = {
```

Comment:
Same for this property.

Reply:
Results from discussion in a separate channel:
- it is correct, as only used in `icon4py` which uses "unstructured" fields. There is max one horizontal dimension per field.
- trace_shifts should work™️ partially on GTIR (but not accross `as_fieldop`).
- currently not actively used in `icon4py`, so defer implementation and put a TODO instead.